corporate/lib/decorator.py

```diff
@@ -123,7 +124,9 @@ def _wrapped_view_func(
                 url = append_url_query_string(url, query)
 
             # Return error for AJAX requests with url.
-            if request.headers.get("x-requested-with") == "XMLHttpRequest":  # nocoverage
+            if (
+                get_preferred_type(request, ["application/json", "text/html"]) != "text/html"
+            ):  # nocoverage
```

Comment:
We should check that our client-side API requests made by self-hosted Zulip servers to this endpoint have `application/json` as what they prefer; maybe just adding a test would suffice?

Reply:
This is only checking that `text/html` is not *strictly more preferred*. This includes a client that sends `Accept: */*` or `Accept: application/json` or `Accept: text/html, application/json` or no `Accept` header at all. It only excludes a client that sends `Accept: text/html, */*;q=0.8` or similar, like browsers do when loading the top-level page.